models/keyword_test.go

```diff
@@ -0,0 +1,123 @@
+package models
+
+import (
+	"testing"
+
+	"github.com/gutakk/go-google-scraper/db"
+	"github.com/gutakk/go-google-scraper/tests"
+	"github.com/stretchr/testify/suite"
+	"gopkg.in/go-playground/assert.v1"
+	"gorm.io/driver/postgres"
+	"gorm.io/gorm"
+)
+
+func TestValidateFileTypeWithValidFileType(t *testing.T) {
+	result := ValidateFileType("text/csv")
+
+	assert.Equal(t, nil, result)
+}
+
+func TestValidateFileTypeWithInvalidFileType(t *testing.T) {
+	result := ValidateFileType("test")
+
+	assert.Equal(t, "File must be CSV format", result.Error())
+}
+
+func TestValidateCSVLengthWithOneRow(t *testing.T) {
+	result := ValidateCSVLength(1)
+
+	assert.Equal(t, nil, result)
+}
+
+func TestValidateCSVLengthWithOneThoudsandRow(t *testing.T) {
+	result := ValidateCSVLength(1000)
+
+	assert.Equal(t, nil, result)
+}
+
+func TestValidateCSVLengthWithZeroRow(t *testing.T) {
+	result := ValidateCSVLength(0)
+
+	assert.Equal(t, "CSV file must contain between 1 to 1000 keywords", result.Error())
+}
+
+func TestValidateCSVLengthWithMinusOneRow(t *testing.T) {
+	result := ValidateCSVLength(-1)
+
+	assert.Equal(t, "CSV file must contain between 1 to 1000 keywords", result.Error())
+}
+
+func TestValidateCSVLengthWithOneThoudsandOneRow(t *testing.T) {
+	result := ValidateCSVLength(1001)
+
+	assert.Equal(t, "CSV file must contain between 1 to 1000 keywords", result.Error())
+}
+
+type KeywordDBTestSuite struct {
+	suite.Suite
+}
+
+func (s *KeywordDBTestSuite) SetupTest() {
+	testDB, _ := gorm.Open(postgres.Open(tests.ConstructTestDsn()), &gorm.Config{})
+	db.GetDB = func() *gorm.DB {
+		return testDB
+	}
+
+	_ = db.GetDB().AutoMigrate(&User{}, &Keyword{})
+}
+
+func (s *KeywordDBTestSuite) TearDownTest() {
+	db.GetDB().Exec("DELETE FROM keywords")
+}
+
+func TestKeywordDBTestSuite(t *testing.T) {
+	suite.Run(t, new(KeywordDBTestSuite))
+}
+
+func (s *KeywordDBTestSuite) TestSaveKeywordsWithValidParams() {
+	keywords := [][]string{
+		[]string{"Hazard"},
+		[]string{"Ronaldo"},
+		[]string{"Neymar"},
+		[]string{"Messi"},
+		[]string{"Mbappe"},
```

Comment:
Based on my previous comments, wouldn't be simpler not to have the nested arrays?

```go
[]string{"Hazard", "Ronaldo", "Neymar", "Messi", "Mbappe"}
```

In the CSV parsing, the data could be cleaned to have this format, no?

Reply:
As I used `ReadAll()` function and the signature look like this
```go
func (r *Reader) ReadAll() (records [][]string, err error) {...}
```

or is it better to loop through file and use `Read()` to construct slice manually instead?

Reply:
I think it is much more better to use `Read()` to create `[]string` slice
Fixed in cd31b5299e42b4d28439db3a244e1cf1d872adae